program/src/processor.rs

```diff
@@ -6667,15 +6663,19 @@ pub fn read_oracle(
 
     let price = match oracle_type {
         OracleType::Pyth => {
-            let price_account = Price::get_price(oracle_ai)?;
+            let oracle_data = oracle_ai.try_borrow_data()?;
+            let price_account = pyth_client::load_price(&oracle_data).unwrap();
             let value = I80F48::from_num(price_account.agg.price);
 
             // Filter out bad prices on mainnet
             #[cfg(not(feature = "devnet"))]
             let conf = I80F48::from_num(price_account.agg.conf).checked_div(value).unwrap();
 
             #[cfg(not(feature = "devnet"))]
-            if price_account.agg.status != PriceStatus::Trading {
+            if price_account.agg.status != PriceStatus::Trading
+                && price_account.last_slot < curr_slot - PYTH_VALID_SLOTS
+            {
+                // Only ignore the pyth price if there hasn't been a valid slot in 50 slots
                 msg!("Pyth status invalid: {}", price_account.agg.status as u8);
                 return Err(throw_err!(MangoErrorCode::InvalidOraclePrice));
```

Comment:
nitpick (only fix if you feel like): I would make this Error mode specific so as to distinguish it from the other InvalidOraclePrice on line:6688